### PR TITLE
[SPARK-11958] [SPARK-11957] [ML] [Doc] SQLTransformer user guide and example code

### PR DESCRIPTION
**Pull request by yanboliang**
Add `SQLTransformer` user guide, example code and make Scala API doc more clear.

### CODE REVIEW

**Comment by yanboliang** [comment]
Jenkins, test this please.

File: R/pkg/DESCRIPTION [hunk 1]
```diff
@@ -35,4 +35,4 @@ Collate:
     'sparkR.R'
     'stats.R'
     'types.R'
-    'utils.R'
+    'utils.R'
```
Comment (mengxr) [inline R/pkg/DESCRIPTION]: not part of this PR

**Comment by mengxr** [comment]
LGTM except one minor issue commented inline.

File: docs/ml-features.md [hunk 2]
```diff
@@ -1591,6 +1591,65 @@ transformer.transform(df).show()
 
 </div>
 
+## SQLTransformer
+
+`SQLTransformer` implements the transforms which are defined by SQL statement.
+Currently we only support SQL syntax like `"SELECT ... FROM __THIS__ ..."`
+where `"__THIS__"` represents the underlying table of the input dataset.
+The select clause specifies the fields, constants, and expressions to display in
+the output, it can be any select clause that Spark SQL supported. Users can also
```
Comment (BenFradet) [inline docs/ml-features.md]: supports and not supported

File: docs/ml-features.md [hunk 3]
```diff
@@ -1591,6 +1591,65 @@ transformer.transform(df).show()
...
+use Spark SQL build-in function and UDFs to operate on these selected columns.
```
Comment (BenFradet) [inline docs/ml-features.md]: built-in

File: docs/ml-features.md [hunk 4]
```diff
@@ -1591,6 +1591,65 @@ transformer.transform(df).show()
...
+For example, `SQLTransformer` support statements like:
```
Comment (BenFradet) [inline docs/ml-features.md]: supports

File: mllib/src/main/scala/org/apache/spark/ml/feature/SQLTransformer.scala [hunk 5]
```diff
@@ -28,8 +28,15 @@ import org.apache.spark.sql.types.StructType
 /**
  * :: Experimental ::
  * Implements the transforms which are defined by SQL statement.
- * Currently we only support SQL syntax like 'SELECT ... FROM __THIS__'
+ * Currently we only support SQL syntax like 'SELECT ... FROM __THIS__ ...'
  * where '__THIS__' represents the underlying table of the input dataset.
+ * The select clause specifies the fields, constants, and expressions to display in
+ * the output, it can be any select clause that Spark SQL supported. Users can also
```
Comment (BenFradet) [inline mllib/src/main/scala/org/apache/spark/ml/feature/SQLTransformer.scala]: supports

File: mllib/src/main/scala/org/apache/spark/ml/feature/SQLTransformer.scala [hunk 6]
```diff
@@ -28,8 +28,15 @@ import org.apache.spark.sql.types.StructType
...
+ * use Spark SQL build-in function and UDFs to operate on these selected columns.
```
Comment (BenFradet) [inline mllib/src/main/scala/org/apache/spark/ml/feature/SQLTransformer.scala]: built-in

**Comment by BenFradet** [comment]
There are a few typos which need to be corrected, but otherwise it looks good.

File: docs/ml-features.md [hunk 7]
```diff
@@ -1591,6 +1591,65 @@ transformer.transform(df).show()
 
 </div>
 
+## SQLTransformer
+
+`SQLTransformer` implements the transforms which are defined by SQL statement.
```
Comment (BenFradet) [inline docs/ml-features.md]: Shouldn't this be "the transformation which are defined by a SQL statement"?

File: mllib/src/main/scala/org/apache/spark/ml/feature/SQLTransformer.scala [hunk 8]
```diff
@@ -28,8 +28,15 @@ import org.apache.spark.sql.types.StructType
 /**
  * :: Experimental ::
  * Implements the transforms which are defined by SQL statement.
```
Comment (BenFradet) [inline mllib/src/main/scala/org/apache/spark/ml/feature/SQLTransformer.scala]: Same here?

File: docs/ml-features.md (hunk 7 above)
Comment (BenFradet) [inline docs/ml-features.md]: Cc @mengxr

**Comment by mengxr** [comment]
Merged into master and branch-1.6. Thanks!
